test/queryBuilders.js

```diff
@@ -0,0 +1,219 @@
+const t = require('../test-lib/test.js');
+const assert = require('assert');
+
+describe('Query Builders', function() {
+  this.timeout(t.timeout);
+
+  let apos;
+  after(function() {
+    return t.destroy(apos);
+  });
+
+  before(async function () {
+    apos = await t.create({
+      root: module,
+      modules: {
+        young: {
+          options: {
+            alias: 'young'
+          },
+          extend: '@apostrophecms/piece-type',
+          fields: {
+            add: {
+              age: {
+                label: 'Age',
+                type: 'integer',
+                required: true
+              }
+            }
+          },
+          queries(self, query) {
+            return {
+              builders: {
+                age: {
+                  launder(str) {
+                    return [ 'children', 'adult' ].includes(str) ? str : null;
+                  },
+                  finalize() {
+                    const age = query.get('age');
+
+                    if ([ 'children', 'adults' ].includes(age)) {
+                      const ageCriteria = age === 'children' ? { $lte: 18 } : { $gt: 18 };
+                      query.and({ age: ageCriteria });
+                    }
+                  }
+                }
+              },
+              methods: {
+                async sortByAge() {
+                  await query.finalize();
+
+                  const pipeline = [
+                    { $match: query.get('criteria') },
+                    { $sort: { age: 1 } }
+                  ];
+
+                  const results = await self.apos.doc.db.aggregate(pipeline).toArray();
+
+                  return results;
+                }
+              }
+            };
+          }
+        },
+        person: {
+          extend: 'young',
+          options: {
+            alias: 'person'
+          },
+          extendQueries(self, query) {
+            return {
+              builders: {
+                age: {
+                  def: 'adult',
+                  launder(_super, str) {
+                    return _super() || str === 'senior';
```

Comment:
This seems wrong. launder returns the value, not a boolean. But your or clause here returns a boolean in the second half, and matches on 'senior' rather than 'seniors'. Something is not right.

Reply:
That's true, my bad.